CRM/Contribute/BAO/Contribution.php

```diff
@@ -4721,15 +4721,27 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values, $recu
       throw new CRM_Core_Exception('Contribution does not exist');
     }
     $contribution->loadRelatedObjects($input, $ids, TRUE);
-    // set receipt from e-mail and name in value
     if (!$returnMessageText) {
-      $session = CRM_Core_Session::singleton();
-      $userID = $session->get('userID');
-      if (!empty($userID)) {
-        list($userName, $userEmail) = CRM_Contact_BAO_Contact_Location::getEmailDetails($userID);
-        $values['receipt_from_email'] = CRM_Utils_Array::value('receipt_from_email', $input, $userEmail);
-        $values['receipt_from_name'] = CRM_Utils_Array::value('receipt_from_name', $input, $userName);
+      // Set receipt from e-mail and name in value if we are sending the message.
+      $userEmail = NULL;
```

Comment:
You don't really need to set these here. They prevent an enotice in theory but in practice there should be no e-notice from the code below so if they DID trigger an e-notice then it would be desirable because something unexpected would have happened